src/client.rs

```diff
@@ -146,6 +146,12 @@ where
         // No more bytes are yielded from the stream.
         assert!(bytes_read != 0, "Empty response"); // TODO: ensure?
 
+        // Prevent CWE-400 DDOS with Large HTTP Headers.
+        ensure!(
+            buf.len() < MAX_HEADER_LENGTH,
```

Comment:
This is the length of the entire buffer which include more than just the headers. Is this the right place for this?

Reply:
Oh yeah, you're right. This would include the initial statuscode + url as well. I'm not sure how useful it's to differentiate between those, given if 8kb worth of HEAD is exceeded it's probably a bad sign either way. But perhaps we should change the variable name to be `MAX_HEAD_LENGTH` since it applies to the full Head section of the HTTP request.

Reply:
Okay, renamed it!